includes/Checker/Checks/Enqueued_Styles_Scope_Check.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\Enqueued_Styles_Scope_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use Exception;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Preparations\Demo_Posts_Creation_Preparation;
+use WordPress\Plugin_Check\Checker\With_Shared_Preparations;
+use WordPress\Plugin_Check\Traits\URL_Aware;
+
+/**
+ * Check if a stylesheet is present in all URLs.
+ *
+ * @since n.e.x.t
+ */
+class Enqueued_Styles_Scope_Check extends Abstract_Runtime_Check implements With_Shared_Preparations {
+
+	use URL_Aware;
+
+	/**
+	 * List of viewable post types.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $viewable_post_types;
+
+	/**
+	 * List of plugin styles.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $plugin_styles = array();
+
+	/**
+	 * Plugin style counter.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $plugin_style_count = array();
+
+	/**
+	 * Runs this preparation step for the environment and returns a cleanup function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Cleanup function to revert any changes made here.
+	 *
+	 * @throws Exception Thrown when preparation fails.
+	 */
+	public function prepare() {
+		$orig_scripts = isset( $GLOBALS['wp_styles'] ) ? $GLOBALS['wp_styles'] : null;
+
+		// Backup the original values for the global state.
+		$this->backup_globals();
+
+		return function() use ( $orig_scripts ) {
+			if ( is_null( $orig_scripts ) ) {
+				unset( $GLOBALS['wp_styles'] );
+			} else {
+				$GLOBALS['wp_styles'] = $orig_scripts;
+			}
+
+			$this->restore_globals();
+		};
+	}
+
+	/**
+	 * Returns an array of shared preparations for the check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Returns a map of $class_name => $constructor_args pairs. If the class does not
+	 *               need any constructor arguments, it would just be an empty array.
+	 */
+	public function get_shared_preparations() {
+		$demo_posts = array_map(
+			static function( $post_type ) {
+				return array(
+					'post_title'   => "Demo {$post_type} post",
+					'post_content' => 'Test content',
+					'post_type'    => $post_type,
+					'post_status'  => 'publish',
+				);
+			},
+			$this->get_viewable_post_types()
+		);
+
+		return array(
+			Demo_Posts_Creation_Preparation::class => array( $demo_posts ),
+		);
+	}
+
+	/**
+	 * Runs the check on the plugin and amends results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check results to amend and the plugin context.
+	 */
+	public function run( Check_Result $result ) {
+		$this->run_for_urls(
+			$this->get_urls(),
```

Comment:
```suggestion
        $urls = $this->get_urls();
		$this->run_for_urls(
			$urls,
```

Get URL, does `WP_Query`, let's cache these in a variable.

Reply:
Updated. Thanks!